Source/SwiftLintBuiltInRules/Rules/Idiomatic/RedundantTypeAnnotationRule.swift

```diff
@@ -1,9 +1,143 @@
+        }
+
+        // Only take the last binding into account in case multiple
+        // variables are declared on a single line.
+        // This binding must have both a type declaration and an initializer
+        // sequence for it to be potentially redundant.
+        guard doesNotContainIgnoredAttributes,
+              let binding = bindings.last,
+              let typeAnnotation = binding.typeAnnotation,
+              let type = typeAnnotation.type.as(IdentifierTypeSyntax.self),
+              let typeName = type.typeName,
+              var initializer = binding.initializer?.value
+        else {
+            return false
+        }
+
+        if let forceUnwrap = initializer.as(ForceUnwrapExprSyntax.self) {
+            initializer = forceUnwrap.expression
+        }
+
+        // If the initializer is a function call (generally a constructor or static builder),
+        // check if the base type is the same as the one from the type annotation.
+        if let functionCall = initializer.as(FunctionCallExprSyntax.self) {
+            if let calledExpression = functionCall.calledExpression.as(DeclReferenceExprSyntax.self) {
+                // Parse generic arguments if there are any.
+                var genericArguments = ""
+                if let genericArgumentsClauseBytes = type.genericArguments?.trimmed.syntaxTextBytes {
+                    genericArguments = String(bytes: genericArgumentsClauseBytes, encoding: .utf8) ?? ""
+                }
+                return calledExpression.baseName.text == typeName + genericArguments
+            }
+
+            // If the function call is a member access expression, check if it is a violation
+            return isMemberAccessViolation(node: functionCall.calledExpression, typeName: typeName)
+        }
+
+        // If the initializer is a boolean expression, we consider using the `Bool` type
+        // annotation as redundant.
+        if initializer.as(BooleanLiteralExprSyntax.self) != nil {
+            return typeName == "Bool"
+        }
+
+        // If the initializer is a member access, check if the base type name is the same as
+        // the type annotation
+        return isMemberAccessViolation(node: initializer, typeName: typeName)
+    }
+
+    /// Checks if the given node is a member access (i.e. an enum case or a static property or function)
+    /// and if so checks if the base type is the same as the given type name.
+    private func isMemberAccessViolation(node: some SyntaxProtocol, typeName: String) -> Bool {
+        guard let memberAccess = node.as(MemberAccessExprSyntax.self),
+              let base = memberAccess.base?.as(DeclReferenceExprSyntax.self) else {
+            // If the type is implicit, `base` will be nil, meaning there is no redundancy.
+            return false
+        }
+
+        return base.baseName.text == typeName
+    }
+}
+
+extension RedundantTypeAnnotationRule {
     static let description = RuleDescription(
```

Comment:
Why is that moved down here? Let's have it right at the beginning after the `configuration` again.